rcl/test/rcl/test_wait.cpp

```diff
@@ -684,3 +685,97 @@ TEST_F(CLASSNAME(WaitSetTestFixture, RMW_IMPLEMENTATION), wait_set_get_allocator
   ret = rcl_wait_set_fini(&wait_set);
   EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
 }
+
+// Test wait set init failure cases using mocks
+TEST_F(CLASSNAME(WaitSetTestFixture, RMW_IMPLEMENTATION), wait_set_failed_init) {
+  rcl_wait_set_t wait_set = rcl_get_zero_initialized_wait_set();
+  // nullptr failures
+  rcl_ret_t ret =
+    rcl_wait_set_init(nullptr, 1, 1, 1, 1, 1, 0, context_ptr, rcl_get_default_allocator());
+  EXPECT_EQ(RCL_RET_INVALID_ARGUMENT, ret);
+  EXPECT_TRUE(rcl_error_is_set());
+  rcl_reset_error();
+
+  ret =
+    rcl_wait_set_init(&wait_set, 1, 1, 1, 1, 1, 0, nullptr, rcl_get_default_allocator());
+  EXPECT_EQ(RCL_RET_INVALID_ARGUMENT, ret);
+  EXPECT_TRUE(rcl_error_is_set());
+  rcl_reset_error();
```

Comment:
@Blast545 shoud this two go to the test case immediately above? Also, should we test using an invalid (zero initialized) allocator?

Reply:
You mean making them part of `wait_set_get_allocator`?

Reply:
Oops, no, the one above that one, `wait_set_valid_arguments`. I misread the diff.

Reply:
Agree, moved with ba06cb9